### PR TITLE
Add missing on error manifest struct destroy in pluginscript script init

### LINKED ISSUE #34041: Pluginscript script init skip manifest destroy on error
In pluginscript initializing a script is done by calling a callback which should return a `godot_pluginscript_script_manifest` structure.
Given this structure contains dictionary and array, it should be destroyed once no longer used:

https://github.com/godotengine/godot/blob/6f38aeef5255d1fdeb99d727a0f67b9be6ccdf36/modules/gdnative/pluginscript/pluginscript_script.cpp#L320-L326

However this is currently not the case if the callback returns an error:

https://github.com/godotengine/godot/blob/6f38aeef5255d1fdeb99d727a0f67b9be6ccdf36/modules/gdnative/pluginscript/pluginscript_script.cpp#L254-L257

I would say forcing the callback to always return a valid manifest seems a cleaner approach (especially considering with the right flags, a C compiler would warn agains returning an initialized variable).

However this change in behaviour may break compatibily (returning an uninitialized manifest cause a segfault).

On top of that this leak is obviously not that of a really big deal because 1) we only leak at most one manifest (so 2 strings, 3 arrays and 1 dictionary) per script whose initialization failed and 2) script initialization failure means user code is obviously buggy (so this should not happened in production)

However it seems [very very few project would be affected](https://github.com/search?q=godot_pluginscript_script_manifest+-extension%3Ah+-filename%3Ascript.py+-filename%3Apluginscript.cpp&type=Code)

In fact only godot-python and godot-ruby would be really affected (godot-clisp is an early WIP).
Given I maintain godot-python, we end up with only godot-ruby, so I guess the change is acceptable ;-)

### PR DESCRIPTION
**Pull request by touilleMan**
fix #34041

### CODE REVIEW

File: modules/gdnative/pluginscript/pluginscript_script.cpp [hunk 1]
```diff
@@ -252,6 +252,13 @@ Error PluginScript::reload(bool p_keep_state) {
 			(godot_string *)&_source,
 			(godot_error *)&err);
 	if (err) {
+		// Manifest's attributes must be explicitly freed
```
Comment (akien-mga) [inline modules/gdnative/pluginscript/pluginscript_script.cpp]: There's another early return at line 279 (`ERR_FAIL_V_MSG`) which would leak similarly I guess?

Comment (touilleMan) [reply modules/gdnative/pluginscript/pluginscript_script.cpp]: good catch ! I've updated the PR ;-)

File: modules/gdnative/pluginscript/pluginscript_script.cpp [hunk 2]
```diff
@@ -317,20 +330,15 @@ Error PluginScript::reload(bool p_keep_state) {
...
-	godot_array_destroy(&manifest.properties);
 
 #ifdef TOOLS_ENABLED
 /*for (Set<PlaceHolderScriptInstance*>::Element *E=placeholders.front();E;E=E->next()) {
 
         _update_placeholder(E->get());
     }*/
 #endif
+
+	FREE_SCRIPT_MANIFEST(manifest);
```
Comment (akien-mga) [inline modules/gdnative/pluginscript/pluginscript_script.cpp]: Please also `#undef FREE_SCRIPT_MANIFEST` after the last use.

Comment (touilleMan) [reply modules/gdnative/pluginscript/pluginscript_script.cpp]: Done !

File: modules/gdnative/pluginscript/pluginscript_script.cpp [hunk 3]
```diff
@@ -251,7 +251,19 @@ Error PluginScript::reload(bool p_keep_state) {
 			(godot_string *)&_path,
 			(godot_string *)&_source,
 			(godot_error *)&err);
+
+	// Manifest's attributes must be explicitly freed
+	#define FREE_SCRIPT_MANIFEST(manifest) { \
```
Comment (akien-mga) [inline modules/gdnative/pluginscript/pluginscript_script.cpp]: `clang-format` doesn't like indented `#define`s: https://travis-ci.org/godotengine/godot/jobs/619222370?utm_medium=notification&utm_source=github_status

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**